### ISSUE TITLE
Issue while running Craft Command

### ISSUE THREAD

**Issue by ghost** [MasoniteFramework/masonite#435]
**Describe the bug**
If you use any command with craft it gives an error.

```
Traceback (most recent call last):
  File "craft", line 12, in <module>
    from wsgi import container
  File "/Users/yubarajshrestha/Python Projects/blog/wsgi.py", line 36, in <module>
    located_provider.load_app(container).register()
  File "/Users/yubarajshrestha/Python Projects/blog/venv/lib/python3.8/site-packages/masonite/providers/AppProvider.py", line 58, in register
    self._autoload(config("application.autoload"))
  File "/Users/yubarajshrestha/Python Projects/blog/venv/lib/python3.8/site-packages/masonite/providers/AppProvider.py", line 67, in _autoload
    Autoload(self.app).load(directories)
  File "/Users/yubarajshrestha/Python Projects/blog/venv/lib/python3.8/site-packages/masonite/autoload.py", line 49, in load
    self._get_module_members(module_loader, name)
  File "/Users/yubarajshrestha/Python Projects/blog/venv/lib/python3.8/site-packages/masonite/autoload.py", line 173, in _get_module_members
    return importlib.import_module(
  File "/usr/local/Cellar/python@3.8/3.8.10/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 122, in import_module
    raise TypeError(msg.format(name))
TypeError: the 'package' argument is required to perform a relative import for '.Users.yubarajshrestha.Python Projects.blog.app.User'
```

**Desktop (please complete the following information):**
 - OS: Mac

**Additional context**
Python Version: 3.8.10
M3 (Masonite v3)

**Comment by girardinsamuel** [comment]
What happen if you remove the space in the folder (Python Projects) ?

**Comment by ghost** [comment]
> What happen if you remove the space in the folder (Python Projects) ?

It was working before, I updated the system this morning and also homebrew, now I encountered this issue.

**Comment by girardinsamuel** [comment]
Oh okay, I will test with 3.8.10 then

**Comment by ghost** [comment]
> Oh okay, I will test with 3.8.10 then

Thanks, the system was perfectly working. I think the issue is with 3.8.10.

**Comment by girardinsamuel** [comment]
Yes 3.8.10 or 3.9.5 which are the latest Python 3 releases from May 3rd.
It has not made its way into pyenv so I was not able to test them with it, I will directly try by installing from python.org

**Comment by girardinsamuel** [comment]
I reproduced the problem locally. I'll try to fix it

**Comment by ghost** [comment]
> I reproduced the problem locally. I'll try to fix it

So this is an issue with 3.8.10.

**Comment by girardinsamuel** [comment]
I guess the issue might happen with 3.9.5 too. No issue with 3.9.4.

**Comment by girardinsamuel** [comment]
`importlib.import_module` has not changed between those releases, so this is due a side-effect somewhere else.
I will look closely at the release log

**Comment by girardinsamuel** [comment]
It's because of `pkgutil` module

Tested in a local masonite project :

### 3.8.10
```python
import pkgutil
for module_loader, name, _ in pkgutil.iter_modules(["app"]):
    print(module_loader, name)
```
> FileFinder('/Users/samuel/repos/masonite/tests/new_3.8.10/app') User

### 3.8.9:
```python
import pkgutil
for module_loader, name, _ in pkgutil.iter_modules(["app"]):
    print(module_loader, name)
```
> FileFinder('app') User

Path is not relative to the current working directory anymore

The detailed changelog of 3.8.10 is here https://docs.python.org/release/3.8.10/whatsnew/changelog.html
I am looking into `pkgutil` and `importlib` related changes

Looks like it could be due to : `Importlib now resolves relative paths when creating module spec objects from file locations`

Related Bug report: https://bugs.python.org/issue43105
PR: https://github.com/python/cpython/pull/25121
Line in the PR where absolute path is built :
https://github.com/python/cpython/commit/0af99b44edf559305def22b2d68be685ce50d7f6#diff-efefe383b3a81d16150c280db0b64eed7569254299418f64cc0d749f8e16f3a4R723

**Comment by girardinsamuel** [comment]
Looks like it could be due to : https://bugs.python.org/issue43105

https://github.com/python/cpython/pull/25121

https://github.com/python/cpython/commit/0af99b44edf559305def22b2d68be685ce50d7f6#diff-efefe383b3a81d16150c280db0b64eed7569254299418f64cc0d749f8e16f3a4R723

**Comment by girardinsamuel** [comment]
@meyubaraj
(can test with `pip install git+https://github.com/girardinsamuel/masonite.git@fix/435`)
@josephmancuso Here is a PR which seems to fix this behaviour. I am not sure if's the way to go but that works.

**Comment by ghost** [comment]
> pip install git+https://github.com/girardinsamuel/masonite.git@fix/435

Not working!

**Comment by girardinsamuel** [comment]
Well it's working on my side (be sure that it haas been correctly installed, maybe run pip uninstall masonite before. And be careful it's a @ not a #, I have edited the comment once).

**Comment by ghost** [comment]
Well, now I am getting a different error. Maybe because of the spaces in my directory structure. Right now, I am thinking to downgrade my python version.

**Comment by josephmancuso** [comment]
🤔 How does Python release a minor version upgrade where they change something from a relative path to an absolute path ?? That's clearly a breaking change, no??

**Comment by ghost** [comment]
Yea, it's weird. I have upgraded to 3.9.4 instead. Now it's working. Please, do close the issue if fixed and the test succeeded. Thanks @girardinsamuel and @josephmancuso

**Comment by girardinsamuel** [comment]
There is a version 3.9.5 released at same time than 3.8.10 which might have the same issue. So we might have to check that a bit more.

**Comment by girardinsamuel** [comment]
> 🤔 How does Python release a minor version upgrade where they change something from a relative path to an absolute path ?? That's clearly a breaking change, no??

I am not 100% sure of what changes those new releases add, but for sure pkgutil.iter_modules behaves differently, and the PR I opened solve this (but maybe it's a hacky patch that we don't want to merge in...).

**Comment by ghost** [comment]
> > 🤔 How does Python release a minor version upgrade where they change something from a relative path to an absolute path ?? That's clearly a breaking change, no??
> 
> I am not 100% sure of what changes those new releases add, but for sure pkgutil.iter_modules behaves differently, and the PR I opened solve this (but maybe it's a hacky patch that we don't want to merge in...).

Any Updates?

**Comment by girardinsamuel** [comment]
Well I did not investigate this no more, another person in Masonite Slack got the same issue with Python 3.8.10.
@josephmancuso what do you think of this patch ?

**Comment by ghost** [comment]
> Well I did not investigate this no more, another person in Masonite Slack got the same issue with Python 3.8.10.
> @josephmancuso what do you think of this patch ?

I think I found a workaround, I request you to give it a try and let me know.

**Comment by ghost** [comment]
> > Well I did not investigate this no more, another person in Masonite Slack got the same issue with Python 3.8.10.
> > @josephmancuso what do you think of this patch ?
> 
> I think I found a workaround, I request you to give it a try and let me know.

@josephmancuso @girardinsamuel  https://github.com/MasoniteFramework/masonite/pull/438

**Comment by josephmancuso** [comment]
@girardinsamuel I merged in your fix.

@yubarajshrestha The fix you made failed almost every test in the autoloading tests so i had to revert the change

**Comment by josephmancuso** [comment]
This is available in version 3.0.12 if you can let me know if that fixes the issue

**Comment by yubarajshrestha** [comment]
I will give it a try and let you know.
